src/tests/unit/tests/injected/adapters/scan-results-to-unified-results.test.ts

```diff
@@ -100,77 +61,58 @@ describe('ScanResults to Unified Results Test', () => {
         const scanResultsStub: ScanResults = createTestResultsWithNoData();
         expect(testSubject.needsReviewConversion(scanResultsStub)).toMatchSnapshot();
     });
 
     test('needsReviewConversion works with filled up passes, failures and incomplete values in scan results', () => {
-        checkResolutionCreatorMock
-            .setup(m =>
-                m({
-                    id: 'test',
-                    nodeResult: passingNode,
-                }),
-            )
-            .returns(() => checkResolutionStub);
-        checkResolutionCreatorMock
-            .setup(m =>
-                m({
-                    id: 'test2',
-                    nodeResult: incompleteNode,
-                }),
-            )
-            .returns(() => checkResolutionStub);
-        checkResolutionCreatorMock
-            .setup(m =>
-                m({
-                    id: 'id1',
-                    nodeResult: node1,
-                }),
-            )
-            .returns(() => checkResolutionStub);
-        checkResolutionCreatorMock
-            .setup(m =>
-                m({
-                    id: 'id1',
-                    nodeResult: node2,
-                }),
-            )
-            .returns(() => checkResolutionStub);
-        checkResolutionCreatorMock
-            .setup(m =>
-                m({
-                    id: 'id2',
-                    nodeResult: node3,
-                }),
-            )
-            .returns(() => checkResolutionStub);
-        const testSubject = new ConvertScanResultsToUnifiedResults(
-            generateGuidMock.object,
-            fixResolutionCreatorMock.object,
-            checkResolutionCreatorMock.object,
+        setupResolutionMock(checkResolutionCreatorMock, 'test', passingNode, checkResolutionStub);
+        setupResolutionMock(checkResolutionCreatorMock, 'id1', node1, checkResolutionStub);
+        setupResolutionMock(checkResolutionCreatorMock, 'id1', node2, checkResolutionStub);
+        setupResolutionMock(checkResolutionCreatorMock, 'id2', node3, checkResolutionStub);
+        setupResolutionMock(
+            checkResolutionCreatorMock,
+            'test2',
+            incompleteNode,
+            checkResolutionStub,
         );
+
+        const testSubject = getTestSubject();
         const scanResultsStub: ScanResults = createTestResultsWithIncompletes();
         expect(testSubject.needsReviewConversion(scanResultsStub)).toMatchSnapshot();
         generateGuidMock.verifyAll();
     });
 
+    function setupResolutionMock(
+        mock: IMock<ResolutionCreator>,
+        id: string,
+        nodeResult: AxeNodeResult,
+        resolutionStub: DictionaryStringTo<any>,
+    ): void {
+        mock.setup(m =>
+            m({
+                id: id,
+                nodeResult: nodeResult,
+            }),
+        ).returns(() => resolutionStub);
+    }
+
+    function getTestSubject(): ConvertScanResultsToUnifiedResults {
```

Comment:
This is good as is but you can also do this in `beforeEach`

Reply:
Gotcha, I thought mocks had to be set up before being used to construct the test object